optuna/multi_objective/samplers/_motpe.py

```diff
@@ -286,20 +286,21 @@ def _split_mo_observation_pairs(
 
             indices_above = np.setdiff1d(indices, indices_below)
 
-            attrs = {"indices_below": indices_below, "indices_above": indices_above}
+            attrs = {
+                "indices_below": indices_below.tolist(),
+                "indices_above": indices_above.tolist(),
+            }
             if self._weights is _default_weights_above:
                 weights_below = self._calculate_default_weights_below(lvals, indices_below)
-                attrs["weights_below"] = weights_below
+                attrs["weights_below"] = weights_below.tolist()
```

Comment:
Should we cast these to lists prior to inserting them into the dict to make the types consistent with the case in which we get a cache hit?

Reply:
Thank you for your comment.
I think these casts are required since `ndarray` is not JSON serializable. I added type casting from `list` to `ndarray` when we deserialize the values from `system_attrs` to make the types consistent (see https://github.com/optuna/optuna/pull/1978/commits/9d0a82bb1e7dc9778995342f00b80945c18e2c93).